tests/components/file_upload/test_init.py

```diff
@@ -64,3 +64,43 @@ async def test_removed_on_stop(hass: HomeAssistant, hass_client, uploaded_file_d
 
     # Test it's removed
     assert not uploaded_file_dir.exists()
+
+
+async def test_upload_large_file(hass: HomeAssistant, hass_client, large_file_io):
+    """Test uploading large file."""
+    assert await async_setup_component(hass, "file_upload", {})
+    client = await hass_client()
+
+    with patch(
+        # Patch temp dir name to avoid tests fail running in parallel
+        "homeassistant.components.file_upload.TEMP_DIR_NAME",
+        file_upload.TEMP_DIR_NAME + f"-{getrandbits(10):03x}",
+    ):
+        res = await client.post("/api/file_upload", data={"file": large_file_io})
+
+    assert res.status == 200
+    response = await res.json()
+
+    file_dir = hass.data[file_upload.DOMAIN].file_dir(response["file_id"])
+    assert file_dir.is_dir()
+
+    large_file_io.seek(0)
+    with file_upload.process_uploaded_file(hass, file_dir.name) as file_path:
+        assert file_path.is_file()
+        assert file_path.parent == file_dir
+        assert file_path.read_bytes() == large_file_io.read().encode("utf-8")
```

Comment:
This puts the whole file in memory twice. Should we do this streaming ?

Reply:
With the file now being really small anyway I would assume we can ignore this?